### DISCUSSION TITLE
Markdown weirdness when using math mode inside `<details>` element

### DISCUSSION THREAD

**Discussion by murari0** [community/community#57950] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

I'm seeing an odd bug when I use a code fenced math block along with a `$` delimited math section inside a `<details>` element:

    <details>
      <summary> Buggy maths </summary>

      $A = 5$
    
      ```math
        A = 5
      ```

    </details>

displays as a regular code block, ignoring the math specification. This only happens if I have a `$` delimited section of math along with the code fenced math block. If I replace the `$A = 5$` with non-math mode text as below, it displays properly

    <details>
      <summary> Works properly </summary>

      non math mode text
    
      ```math
        A = 5
      ```

    </details>

<details>
  <summary> Buggy maths </summary>

  $A = 5$
  
  ```math
    A = 5
  ```

</details>

<details>
  <summary> Works properly </summary>

  non math mode text
  
  ```math
    A = 5
  ```

</details>

This only happens inside `<details>` elements as far as I have tested. I'm not very familiar with Markdown in general - is this a Github bug? I know display math is a Github specific extension but I don't know if this is a bug in `<details>` or in the math environment.

**Comment by nschloe** [comment]
Good catch! I've added it to my bug list at https://github.com/nschloe/github-math-bugs#inline-and-display-math-in-details. (As you can see, still lots of things not working properly.)

**Comment by amrzv** [comment]
Bumping this!

It's annoying, especially when it renders in vscode but [doesn't work](https://github.com/amrzv/olymp/blob/2006-IV/2006-2007/IV/8.md) on github.

![image](https://github.com/user-attachments/assets/b0a62acd-12cf-4baf-83bd-3f81f9070641)
